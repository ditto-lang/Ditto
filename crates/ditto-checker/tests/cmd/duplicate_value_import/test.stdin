module Test exports (..)

import Foo (five, five)

fives = [five, five, five]
