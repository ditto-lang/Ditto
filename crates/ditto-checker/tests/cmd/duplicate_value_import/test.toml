bin.name = "ditto-checker-testbin-check-module"
args = ["duplicate_value_import.ditto"]
fs.sandbox = true
