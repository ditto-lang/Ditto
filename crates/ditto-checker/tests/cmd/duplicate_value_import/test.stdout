{
  "module_name": [
    "Test"
  ],
  "exports": {
    "types": {},
    "constructors": {},
    "values": {
      "fives": {
        "doc_comments": [],
        "doc_position": 0,
        "value_type": {
          "type": "Call",
          "data": {
            "function": {
              "type": "PrimConstructor",
              "data": "Array"
            },
            "arguments": [
              {
                "type": "PrimConstructor",
                "data": "Int"
              }
            ]
          }
        }
      }
    }
  },
  "types": {},
  "constructors": {},
  "values": {
    "fives": {
      "doc_comments": [],
      "name_span": {
        "start_offset": 53,
        "end_offset": 58
      },
      "expression": {
        "expression": "Array",
        "data": {
          "span": {
            "start_offset": 61,
            "end_offset": 79
          },
          "element_type": {
            "type": "PrimConstructor",
            "data": "Int"
          },
          "elements": [
            {
              "expression": "ImportedVariable",
              "data": {
                "span": {
                  "start_offset": 62,
                  "end_offset": 66
                },
                "variable_type": {
                  "type": "PrimConstructor",
                  "data": "Int"
                },
                "variable": {
                  "module_name": [
                    null,
                    [
                      "Foo"
                    ]
                  ],
                  "value": "five"
                }
              }
            },
            {
              "expression": "ImportedVariable",
              "data": {
                "span": {
                  "start_offset": 68,
                  "end_offset": 72
                },
                "variable_type": {
                  "type": "PrimConstructor",
                  "data": "Int"
                },
                "variable": {
                  "module_name": [
                    null,
                    [
                      "Foo"
                    ]
                  ],
                  "value": "five"
                }
              }
            },
            {
              "expression": "ImportedVariable",
              "data": {
                "span": {
                  "start_offset": 74,
                  "end_offset": 78
                },
                "variable_type": {
                  "type": "PrimConstructor",
                  "data": "Int"
                },
                "variable": {
                  "module_name": [
                    null,
                    [
                      "Foo"
                    ]
                  ],
                  "value": "five"
                }
              }
            }
          ],
          "value_type": {
            "type": "Call",
            "data": {
              "function": {
                "type": "PrimConstructor",
                "data": "Array"
              },
              "arguments": [
                {
                  "type": "PrimConstructor",
                  "data": "Int"
                }
              ]
            }
          }
        }
      }
    }
  },
  "values_toposort": [
    "fives"
  ]
}